test/unittests/t_util.cc

```diff
@@ -209,3 +307,853 @@ TEST(T_Util, IsHttpUrl) {
+  close(fd1);
+  close(fd2);
+  close(fd3);
+  close(fd4);
+}
+
+TEST_F(T_Util, Trim) {
+  EXPECT_EQ("", Trim(""));
+  EXPECT_EQ("hello", Trim("  hello"));
+  EXPECT_EQ("hello", Trim("        hello    "));
+  EXPECT_EQ("he llo how are you", Trim("    he llo how are you   "));
+  EXPECT_EQ("hell o", Trim("  hell o"));
+}
+
+TEST_F(T_Util, ToUpper) {
+  EXPECT_EQ("", ToUpper(""));
+  EXPECT_EQ("HELLO", ToUpper("hello"));
+  EXPECT_EQ("HELLO", ToUpper("Hello"));
+  EXPECT_EQ("HELLO HOW ARE YOU?", ToUpper("hElLo HOW are yOu?"));
+  EXPECT_EQ(" 123 ", ToUpper(" 123 "));
+  EXPECT_EQ("HELLO123 456", ToUpper("hEllo123 456"));
+}
+
+TEST_F(T_Util, ReplaceAll) {
+  EXPECT_EQ("atlas.cern.ch", ReplaceAll("@repo@.cern.ch", "@repo@", "atlas"));
+  EXPECT_EQ("aa", ReplaceAll("a@remove@a", "@remove@", ""));
+  EXPECT_EQ("mystring", ReplaceAll("mystring", "", ""));
+  EXPECT_EQ("??? ???my?string???", ReplaceAll("... ...my.string...", ".", "?"));
+  EXPECT_EQ("it's for REPLACED or for @not-replace@",
+      ReplaceAll("it's for @replace@ or for @not-replace@", "@replace@",
+          "REPLACED"));
+}
+
+TEST_F(T_Util, BlockSignal) {
+  EXPECT_DEATH(kill(getpid(), SIGUSR1), ".*");
+  BlockSignal(SIGUSR1);
+  kill(getpid(), SIGUSR1);  // it doesn't crash after blocking
+  EXPECT_DEATH(BlockSignal(-2000), ".*");
+}
+
+TEST_F(T_Util, WaitForSignal) {
+  tbb::tbb_thread signal_sender(SendSignal, SIGUSR1);
+  WaitForSignal(SIGUSR1);
+  signal_sender.join();
+}
+
+TEST_F(T_Util, Daemonize) {
+  int pid;
+  int statloc;
+  int child_pid;
+  int pipe_pid[2];
+  MakePipe(pipe_pid);
+  if ((pid = fork()) == 0) {
+    Daemonize();
+    child_pid = getpid();
+    WritePipe(pipe_pid[1], &child_pid, sizeof(int));
+    _exit(0);
+  } else {
+    waitpid(pid, &statloc, 0);
+    ReadPipe(pipe_pid[0], &child_pid, sizeof(int));
+    EXPECT_NE(getpid(), child_pid);
+    ClosePipe(pipe_pid);
+  }
+}
+
+TEST_F(T_Util, ExecuteBinary) {
+  int fd_stdin;
+  int fd_stdout;
+  int fd_stderr;
+  char buffer[20];
+  bool result;
+  string message = "CVMFS";
+  vector<string> argv;
+  argv.push_back(message);
+  pid_t gdb_pid = 0;
+
+  result = ExecuteBinary(
+      &fd_stdin,
+      &fd_stdout,
+      &fd_stderr,
+      "/bin/echo",
+      argv,
+      false,
+      &gdb_pid);
+  EXPECT_TRUE(result);
+  read(fd_stdout, buffer, message.size());
+  string response(buffer);
+  EXPECT_EQ(message, response);
+}
+
+TEST_F(T_Util, Shell) {
+  int fd_stdin;
+  int fd_stdout;
+  int fd_stderr;
+  const int buffer_size = 100;
+  char buffer[buffer_size];
+
+  EXPECT_TRUE(Shell(&fd_stdin, &fd_stdout, &fd_stderr));
+  string path = sandbox + "/" + "newfolder";
+  string command = "mkdir -p " + path +  " && cd " + path + " && pwd\n";
+  WritePipe(fd_stdin, command.c_str(), command.length());
+  ReadPipe(fd_stdout, buffer, path.length());
+  string result(buffer);
+
+  EXPECT_EQ(path, result);
+  close(fd_stdin);
+  close(fd_stdout);
+  close(fd_stderr);
+}
+
+TEST_F(T_Util, ManagedExec) {
```

Comment:
To my own little surprise, I found [this](https://github.com/cvmfs/cvmfs/blob/devel/test/unittests/t_managed_exec.cc). O.o I actually jotted together a couple of unit tests for `ManagedExec` back in 2013.